static/js/publisher/release/components/releasesTable/index.js

```diff
@@ -67,7 +87,117 @@ class ReleasesTable extends Component {
   }
 
   renderHistoryPanel() {
-    return <HistoryPanel key="history-panel" />;
+    return (
+      <div className="p-releases-table__row" key="history-panel-row">
+        <div className="p-releases-channel is-placeholder u-hide--small" />
+        <HistoryPanel key="history-panel" />
+      </div>
+    );
+  }
+
+  renderAvailableRevisionsRow() {
+    const { isHistoryOpen, filters } = this.props;
+
+    return (
+      <Fragment>
+        {this.renderChannelRow(AVAILABLE)}
+        {isHistoryOpen &&
+          filters.risk === AVAILABLE &&
+          this.renderHistoryPanel()}
+      </Fragment>
+    );
+  }
+
+  renderAvailableRevisions() {
+    return (
+      <Fragment>
+        <h5>Promote from uploaded revisions</h5>
+
+        {this.renderAvailableRevisionsRow()}
+      </Fragment>
+    );
+  }
+
+  renderLaunchpadBuilds() {
+    const lpRevisions = this.props.launchpadRevisions;
+    const { showAllBuilds } = this.state;
+
+    const builds = lpRevisions
+      .map(getBuildId)
+      .filter((item, i, ar) => ar.indexOf(item) === i)
+      .map(buildId => {
+        const revs = this.props.getRevisionsFromBuild(buildId);
+
+        const revsMap = {};
+
+        revs.forEach(r => {
+          r.architectures.forEach(arch => {
+            revsMap[arch] = r;
+          });
+        });
+
+        return revsMap;
+      });
+
+    let buildsToShow = builds;
+
+    if (!showAllBuilds) {
+      buildsToShow = builds.slice(0, MAX_BUILDS);
+    }
+
+    if (builds.length) {
+      return (
+        <Fragment>
+          <h5>Promote from recent Launchpad builds</h5>
+          {buildsToShow.map(revisions => this.renderBuildRow(revisions))}
+
+          {builds.length > MAX_BUILDS && (
+            <div className="p-releases-table__row--show-all">
+              <a onClick={this.handleToggleShowMoreBuilds.bind(this)}>
```

Comment:
We could maybe think about making this into a wrapping component at some point, a "Show all x (y)" and "Show only z x". We have something similar for the branches.